infrastructure/data.tf

```diff
@@ -17,3 +17,17 @@ data "azurerm_monitor_action_group" "common" {
   resource_group_name = var.common_config.resource_group_name
   name                = each.value
 }
+
+data "azurerm_cdn_frontdoor_profile" "web" {
+  name                = var.front_door_config.frontdoor_name
+  resource_group_name = var.front_door_config.frontdoor_rg
+  provider            = azurerm.tooling
```

Comment:
This bit will have to be optional. I think there will be a way to do it, but not 100% sure.

Perhaps: 
```
provider            = var.front_door_config.use_tooling ? azurerm.tooling : undefined
```
(would require an extra variable in the config to set `use_tooling` to true or false)

This would apply to all the provider lines in the files - so it may be best to make it a local variable so it can be referenced, e.g. in locals:

`front_door_provider = var.front_door_config.use_tooling ? azurerm.tooling : undefined`